docs/notebooks/Super Dense Coding.ipynb

```diff
@@ -0,0 +1,259 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2024-11-20T12:37:04.536147800Z",
+     "start_time": "2024-11-20T12:37:04.456110900Z"
+    },
+    "collapsed": false,
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "import matplotlib.pyplot as plt\n",
+    "from qiskit import QuantumCircuit\n",
+    "\n",
+    "from qiskit_quantuminspire.qi_provider import QIProvider"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": false,
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "source": [
+    "### Generate the circuit\n",
+    "\n",
+    "The two qubits are initially prepared in the ground state 00. Before sending the qubit to Bob, Alice needs to prepare an entangled state. So, Alice applies a H-gate followed by a CNOT-gate to obtain the desired entangled state (1/sqrt(2)(00 + 11). Now the qubit is ready to be sent to Bob, who encodes the two classical bits of information.\n",
+    "\n",
+    "Bob can choose between four gates to apply to the qubit. Each gate will encode a different message for Alice. The four different possibilities are listed in the table below.\n",
+    "\n",
+    "| Gates\t | Classical Message |\n",
+    "|:------:| :------: |\n",
+    "|   I\t   | 00 |\n",
+    "|   X\t   | 01 |\n",
+    "|   Z\t   | 10 |\n",
+    "|   ZX   | 11 |\n",
+    "\n",
+    "A different Bell state will be encoded for each gate that Bob applies to the qubit. After decoding, each Bell state will result in a different 2-bit message. Remember that the Identity gate doesn't alter the state, and the X and Z gates make the qubit do a pi-rotation over their respective axes.\n",
+    "\n",
+    "The different possible states after encoding will then be:\n",
+    "\n",
+    "After the encoding step, Bobs sends the qubit back to Alice. To decode the message, Alice applies a CNOT-gate followed by a H-gate, see Figure 1. The decoding of the '11' message is: H CNOT (1/sqrt(2)(-10 + 01) = 11\n",
+    "\n",
+    "The states after the different respective gate operations are listed in Table 1."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "ExecuteTime": {
+     "start_time": "2024-11-20T11:02:48.197216900Z"
+    },
+    "collapsed": false,
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "def superdense_encoding_circuit(bit_a: int =0, bit_b: int =0) -> QuantumCircuit:\n",
+    "    circuit = QuantumCircuit(2, 2)\n",
+    "    qubit_a = 0\n",
+    "    qubit_b = 1\n",
+    "    #circuit.reset(qubit_a)\n",
```

Comment:
reset works on the new qxelarator. So maybe uncomment?

Reply:
Yes.